skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -78,6 +79,9 @@ protected void onCreate(Bundle savedInstanceState) {
         setTitle(getString(R.string.connect_bluetooth_title));
         setSupportActionBar(toolbar);
 
+        //checking the location permission for bluetooth for android 9.
+        PermissionUtils.requestLocationPermission(this);
```

Comment:
If it's required for only Android 9 devices, then I think we should not ask for this permission unnecessarily for other devices in which it is not required.

Reply:
Other devices don't need to ask for permission because it was granted by default but we don't know if there are something unexpected. The code has `if-else` to check so I think this more insurance, what do you think? @lakshyagupta21

Reply:
I can remove that comments.

Reply:
How the permissions are granted by default, I'm asking this because `Location` is considered as [dangerous permission](https://developer.android.com/guide/topics/permissions/overview#permission-groups) and for devices greater than Android 5 apps should ask this at runtime to user, otherwise app doesn't have this permission.


> Other devices don't need to ask for permission

Are you talking about devices with versions lower than 5?

Reply:
Yes, but as for the devices in android 5-8, the application will give that permission without asking, this is why sometimes we don't need to ask for that. Anyway, in this change, the location permission is required for bluetooth, if the location didn't turned on, we can discover nothing. I added the `if` check to see if our application granted this permission, if not we will popup the dialog to request, so I think we don't have to set the specific API code (like target API is android 9), just call this `PermissionUtils.requestLocationPermission(this)`.

Reply:
I think you noticed that `Location` permission is required in Android 9 so that it can connect to bluetooth or to discover the bluetooth devices. Theoretically, app should not be granted `Dangerous` permission automatically if the device used is on version >= 5,0 and if the compileSDKVersion is also greater than or equal to 23. I just checked in Android 7.1 that it doesn't need any location permission to play with `Bluetooth`. And Bluetooth is also enabled even the location permission is not given to the app. 
Correct me if I'm wrong at any point.

Reply:
You are right, I'll update the code. Thanks for your explanation.